src/main/kotlin/oop/Iterator/Iterable.kt

```diff
@@ -0,0 +1,6 @@
+package oop.Iterator
+
+interface Iterable<T> {
+  fun getIterator(type: TypeIterator): Iterator<T>
```

Comment:
If you are going to have several Iterator types is it better to create these iterators this way (by passing a parameter with the desired type as a flag) or by offering a method for each Iterator you are going to have? @tonilopezmr what do you think? 🤔